### PR TITLE
[monaco] Apply multiple changes per document at once (fixes #1049)

### LINKED ISSUE #1049: WorkspaceEdit is not applied correctly
When a `WorkspaceEdit` has multiple `TextEdits` they are not applied correctly. This is probably because they are applied one after the other while all the `TextEdit`s on a `WorkspaceEdit` is relative to the initial state. 

Also undo for a `WorkspaceEdit` should undo all `TextEdits` included on `WorkspaceEdit` but it undoes only one `TextEdit` at a time.

### PR DESCRIPTION
**Pull request by svenefftinge**
Signed-off-by: Sven Efftinge <sven.efftinge@typefox.io>

### CODE REVIEW

File: packages/monaco/src/browser/monaco-workspace.ts [hunk 1]
```diff
@@ -144,32 +145,46 @@ export class MonacoWorkspace extends BaseMonacoWorkspace implements lang.Workspa
...
-                }
+                }));
+                // start a fresh operation
+                model.pushStackElement();
+                model.pushEditOperations([currentSelection], editOperations, (undoEdits: monaco.editor.IIdentifiedSingleEditOperation[]) => [currentSelection]);
                 // push again to make this change an undoable operation
                 model.pushStackElement();
-                reference.dispose();
-            }));
+                editor.focus();
```
Comment (akosyakov) [inline packages/monaco/src/browser/monaco-workspace.ts]: I am not sure what will happen if an editor is not visible. You can also pass `revealIfVisible` to `open`.

Comment (svenefftinge) [reply packages/monaco/src/browser/monaco-workspace.ts]: I will remove the `focus` completely. I don't think applying an edit should do anything the focus. E.g. you do a complex refactoring, you want to stay where you are.

File: packages/monaco/src/browser/monaco-workspace.ts [hunk 2]
```diff
@@ -144,32 +145,46 @@ export class MonacoWorkspace extends BaseMonacoWorkspace implements lang.Workspa
...
-                const range = edit.range;
-                const selections = [new monaco.Selection(range.startLineNumber, range.startColumn, range.endLineNumber, range.endColumn)];
-                model.pushEditOperations(selections, [{
+        const uri2Edits = this.groupEdits(workspaceEdit);
+        for (const uri of uri2Edits.keys()) {
+            const editorWidget = await this.editorManager.open(new URI(uri));
+            const editor = get(editorWidget);
+            if (editor) {
+                const model = editor.document.textEditorModel;
+                const currentSelection = new monaco.Selection(
```
Comment (akosyakov) [inline packages/monaco/src/browser/monaco-workspace.ts]: `const currentSelections = editor.editor.getSelections();`

Comment (svenefftinge) [reply packages/monaco/src/browser/monaco-workspace.ts]: `editor.editor` is protected

Comment (akosyakov) [reply packages/monaco/src/browser/monaco-workspace.ts]: sorry, I meant `editor.getControl().getSelections()`

**Pull request merged**